data/lua/core/wml.lua

```diff
@@ -359,6 +362,49 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 			wml.array_access.set(key, value)
 		end
 	})
+	
+	--[========[Global persistent variables]========]
```

Comment:
This is code moving from World Conquest, but is it generally useful code, and is it presenting the right API for general use? I don't think it is, I think this should be left in WC for the moment, and left for when there's time to properly design an API for 1.17.

* It doesn't seem to have any documentation on the Wiki yet.
* The `immediate=true` makes sense to me, but means it doesn't provide all of the documentated behavior of the WML tag.
* While this might just be that I'm not used to Lua coding conventions, the way that the namespace is accessed via `global_vars.wc2.skip_pickup_dialog` feels a bit odd.

Reply:
I moved it because I thought it _did_ look like a good, generally useful API. It's true that it's not exposing the transactional nature of PersistenceWML though…

Reply:
I feel we'll want to change the interface again during 1.17, and as the existing code is campaign-specific it doesn't need to be in the "get all the deprecations done now so that there aren't more deprecations in 1.17" group.

Why rush to make this a general API in 1.16 instead of waiting until we have time to design it slowly?

Reply:
It's already a general API in 1.14, so your argument doesn't really make sense…

Reply:
What was already a general API in 1.14? The Lua code is being copied from World Conquest, so it wasn't general Lua API before.

Reply:
I don't remember my thoughts when I originally posted that, but I may have been referring to the WML API?

What about putting this in the "experimental" module that Pentarctagon suggested? Then it can be generally available in 1.16 but also more easily iterated on if we decide on a better design.

Reply:
As it doesn't need any C++ changes, I don't see what it gives to UMC authors. My alternative would be for UMC authors in 1.14/1.16 to just cut-and-paste the block from WC into their own add-on's utils files, which will still work unchanged in 1.18 even if we add a nicer API for 1.18.

Reply:
It's definitely much more visible.  As it is, the only way any UMC author would know to do that would be if they found the functionality themselves somehow or were told by someone else who had found it.  It also provides an opportunity for people to provide feedback on something that will probably become a non-experimental API at some point.

Reply:
Alright, I moved it to experimental.